evalml/pipelines/components/estimators/regressors/prophet_regressor.py

```diff
@@ -146,9 +146,12 @@ def predict(self, X, y=None):
             X=X, y=y, date_column=self.parameters["date_index"]
         )
 
-        y_pred = self._component_obj.predict(prophet_df)["yhat"]
-        y_pred = y_pred.rename(None)
-        return y_pred
+        prophet_output = self._component_obj.predict(prophet_df)
+        predictions = prophet_output["yhat"]
+        predictions = infer_feature_types(predictions)
+        predictions = predictions.rename(None)
+        predictions.index = prophet_output["ds"]
```

Comment:
Reading the docs [here](https://facebook.github.io/prophet/docs/quick_start.html#python-api), it looks like the date index information is stored in "ds".